code/game/objects/items/tools/shovel_tools.dm

```diff
@@ -200,4 +204,24 @@
 	. = ..()
 	if(sharp)
 		to_chat(user, "<span class='notice'> This one has been sharpened and can no longer be folded.</span>")
+/obj/item/tool/shovel/unique_action(mob/user)
+	entrenchingmode = !entrenchingmode
+	if(entrenchingmode)
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to entrenching mode.</span>")
+		return
+	else
+		user.visible_message("<span class='notice'>[user] reconfigures [src] to digging mode.</span>")
+		return
 
+///Digs a trench if there is no trench in that tile already.
+/obj/item/tool/shovel/proc/dig_trench(turf/target, mob/user)
+	var/obj/structure/B = locate() in target
+	if(B)
+		if(!istype(B, /obj/structure/cable))
+			to_chat(user, "<span class='notice'>You cannot place a trench there!</span>")
+			return
```

Comment:
You can only build on cables?
Also name this var better please B is not very indicative of what this is supposed to be

Reply:
No, you can only build on floor and cables, but not walls, or pipe

Reply:
Renamed B